### ISSUE TITLE
Errors in MSSQL migrate

### ISSUE THREAD

**Issue by pokrad** [lucent-sea/Remotely#284]
HI !

I tried to build and deploy server to IIS on windows from git sources. When using SqlLite database provider everything works fine.

But when using MSSQL database provider,  context.Database.Migrate();

throws exception:

{"To change the IDENTITY property of a column, the column needs to be dropped and recreated."}

The error is these two statements in "20210408133030_Add ScriptSchedules"

            migrationBuilder.AlterColumn<int>(
                name: "Id",
                table: "AspNetUserClaims",
                type: "int",
                nullable: false,
                oldClrType: typeof(int),
                oldType: "int")
                .Annotation("SqlServer:Identity", "1, 1");

            migrationBuilder.AlterColumn<int>(
                name: "Id",
                table: "AspNetRoleClaims",
                type: "int",
                nullable: false,
                oldClrType: typeof(int),
                oldType: "int")
                .Annotation("SqlServer:Identity", "1, 1");

I am not sure why those two AlterColumn statements are there, because in 

"20201002214839_InitialCreate" ID column is already "int not null identity(1,1)"

            migrationBuilder.CreateTable(
                name: "AspNetUserClaims",
                columns: table => new
                {
                    Id = table.Column<int>(nullable: false)
                        .Annotation("SqlServer:ValueGenerationStrategy", SqlServerValueGenerationStrategy.IdentityColumn),
                    UserId = table.Column<string>(nullable: false),
                    ClaimType = table.Column<string>(nullable: true),
                    ClaimValue = table.Column<string>(nullable: true)
                },
                constraints: table =>
                {
                    table.PrimaryKey("PK_AspNetUserClaims", x => x.Id);
                    table.ForeignKey(
                        name: "FK_AspNetUserClaims_RemotelyUsers_UserId",
                        column: x => x.UserId,
                        principalTable: "RemotelyUsers",
                        principalColumn: "Id",
                        onDelete: ReferentialAction.Cascade);
                });

but I am not that good with EF, so someone with more knowledge might take a look.
Commenting out those two statements makes migrate pass, and I checked, both of those two columns has identity in the database...

Also, one more comment about migrate:
When deploying application I rather have server throw exception if database migrate fails than just log the error and continue.
That was very confusing as I had problems figuring out where the problem actually was. Error was thrown later in application saying that I do not have "ScriptsRun" table created... It took some time to get the actual error. Also if migrate does not pass succesffully, and You continue, theoretically Your new server could mess up the data...
So I suggest adding throw to the migrate part as follows:

            try
            {
                if (context.Database.IsRelational())
                {
                    context.Database.Migrate();
                }
            }
            catch (Exception ex)
            {
                Console.WriteLine(ex.Message);
                throw;
            }

Again, I do not use EF, so this is just a suggestion, I am not familiar with the "Remotely" internals yet :)

Regards !

**Comment by lucent-sea** [comment]
Thanks for finding this!  Not a lot of people use Windows, and it's awesome to get such a detailed bug report.

Those two lines keep getting generated automatically by EF Core's migration builder.  I thought I fixed it when I upgraded my EF Core tools on my local machine.  But after testing it again now, I think it was happening because the SQL Server DB in my LocalDb had an old schema in it.  Seems to be fixed after deleting the database and letting EF Core rebuild it.

Anyhow!  I deleted those lines from the migration.  And I'll look out for this in future migrations, just in case.

I also agree with you about the try/catch around `Migrate`.  I'll remove it.  :)

I'll post again when these are published in a new pre-release.
